src/Highlight.ts

```diff
@@ -3,51 +3,63 @@ import SerializedHighlight from './SerializedHighlight';
 
 export const FOCUS_CSS = 'focus';
 
-export default class Highlight {
-  private _id: string;
-  private _content: string;
-  private _range: Range;
-  private _elements: HTMLElement[] = [];
+export interface IHighlightData {
+  style?: string;
+  id: string;
+  content: string;
+}
 
-  constructor(range: Range, content: string);
-  constructor(id: string, range: Range, content: string);
-  constructor(arg: any, arg2: any, content?: string) {
-    if (arg instanceof Range) {
-      // TODO - something more random here
-      this._id = (new Date()).getTime().toString();
-      this._range = arg;
-      this._content = arg2;
-    } else if (content) {
-      this._id = arg;
-      this._range = arg2;
-      this._content = content;
-    } else {
-      throw new Error('invalid constructor arguments');
-    }
-  }
+export default class Highlight {
 
   public get id(): string {
-    return this._id;
+    return this.data.id;
   }
 
   public get content(): string {
-    return this._content;
-  }
-
-  public get range(): Range {
-    return this._range;
+    return this.data.content;
   }
 
   public set elements(elements: HTMLElement[]) {
     if (this.elements.length > 0) {
       throw new Error(`Hightlight elements aren't reloadable`);
     }
     this._elements = elements;
+    this.loadStyle();
   }
 
   public get elements(): HTMLElement[] {
     return this._elements;
   }
+  public readonly range: Range;
+  private data: IHighlightData;
+  private _elements: HTMLElement[] = [];
+
+  constructor(range: Range, data: Pick<IHighlightData, Exclude<keyof IHighlightData, 'id'>> & Partial<Pick<IHighlightData, 'id'>>) {
+    this.range = range;
+    this.data = {
+      ...data,
+      // TODO - something more random here
+      id: data.id || (new Date()).getTime().toString(),
+    };
+  }
+
+  public setStyle(style: string) {
+    this.removeStyle();
+    this.data.style = style;
+    this.loadStyle();
+  }
+
```

Comment:
it's probably no big deal, since it's unlikely the highlighter wwill have 100's of highlights.

This loops through and manipulates elements twice, which might cause minor thrashing.  That could be avoided by writing the `add` & `remove` as functions that take an element, then loop through and call the appropriate functions in each of these methods.

Reply:
you're right, but it hasn't been an issue so i'll defer the change for now